### ISSUE TITLE
Highlight does not mostly work

### ISSUE THREAD

**Issue by jarmo** [elixir-editors/vim-elixir#452]
# Actual
Highlight does not seem to work mostly. For example, this `exs` has no highlight at all:

<img width="426" alt="image" src="https://user-images.githubusercontent.com/174160/48262304-af8fdb00-e42a-11e8-80d3-0256177a919a.png">

This has only very few highlighting:
<img width="449" alt="image" src="https://user-images.githubusercontent.com/174160/48262367-ecf46880-e42a-11e8-876d-d79552d884c0.png">

`*.html.eex` does not seem to have any highlighting if there's no Elixir code in there (only html or JavaScript, for example), but seems to have higlighting if there's at least some Elixir code inside.

# Expected

To have a lot of highlighting instead of only little.

Am I doing something wrong?

**Comment by jbodah** [comment]
Some thoughts:

* do you have `syntax on` set? 
* do you have a colorscheme set?
* does your colorscheme work for other languages using the same terminal emulator?

If none of those work then I have a few more suggestions for debugging

**Comment by jbodah** [comment]
So the highlighting works in two phases:

* First, the syntax regexs in [the syntax file](https://github.com/elixir-editors/vim-elixir/blob/master/syntax/elixir.vim) map the code into syntax groups. The convention seems to be to map them to language-specific syntax groups and then to link them to more a set of common built-in groups

* Second, the colorscheme is responsible for defining how to color ("highlight") each syntax group. These could be language-specific groups, or the high-level common groups

First I think we need to check that the syntax groups are being properly assigned. I've had good luck doing the following:

1. Put the vim cursor on some piece of code
2. Run `:echo map(synstack(line('.'), col('.')), 'synIDattr(v:val, "name")')`
3. That should output the stack of syntax groups for that position. If you don't see something like `['elixirBlock', ...a bunch more stuff...]` then syntax matching likely isn't working

If the syntax looks correct, the next step would be to debug the highlighting. This will depend on your specific colorscheme. You can either open the colorscheme you're using and check the code for it, or you can call `:hi` to view all of the highlight mappings or `:hi mySyntaxGroup` to inspect a specific group

Are you using a public colorscheme that I could pull down?

**Comment by jarmo** [comment]
Hi!

Thank You for your quick and insightful help. I've verified that syntax
groups aren't properly assigned for Elixir (but they are for Ruby). Only
things which have color in my Elixir code will get assigned syntax groups
for something called "euphoria3". How to troubleshoot further?

Colorscheme is newspaper - I think that it is builtin? But thanks for
opening my eyes - I've never thought that colorscheme might have less
highlighting built in.

Jarmo

**Comment by jbodah** [comment]
Do you have vim-elixir explicitly installed? Note it is not bundled with vim by default like vim-ruby is

**Comment by jarmo** [comment]
Yes, it is installed via Vundle as:
```
Plugin 'elixir-editors/vim-elixir'
```

**Comment by jbodah** [comment]
I'm guessing that the filetype is not being properly recognized and thus the autoloading is not working properly. What does `:filetype` show you? What about `:set filetype?`

**Comment by jarmo** [comment]
`filetype detection:ON plugin:ON indent:ON` and `filetype=euphoria3` so yeah it seems that last part might be the culprit here. I don't have any clue what `euphoria3` is and how it has gotten there. Is there any way for me to force Vim to think that these files are Elixir instead?

**Comment by jarmo** [comment]
I've managed to make syntax highlighting work as expected with adding these lines to vim `.vimrc`:
```vim
autocmd BufNewFile,BufRead *.ex,*.exs set syntax=elixir
autocmd BufNewFile,BufRead *.eex set syntax=eelixir
```
I can see elixir specific stack of syntax groups and actual highlighting.

If you have any ideas how to troubleshoot further initial problem so that editing `.vimrc` would not be needed then I'm happy to try these suggestions out.

**Comment by jbodah** [comment]
Just dug up this old issue: https://github.com/elixir-editors/vim-elixir/issues/121

Could you post your vimrc as well as `vim --version`? I'm having trouble repro'ing with `vim -u NONE`

**Comment by jarmo** [comment]
That issue helped me! Thanks!

I've managed to fix this whole problem (I don't need au commands above anymore). This is how my `.vimrc` looked like (shortened of course):

```vim
filetype on

set rtp+=~/.vim/bundle/Vundle.vim/
call vundle#begin()
Plugin 'VundleVim/Vundle.vim'
"...more plugins
Plugin 'elixir-editors/vim-elixir'
call vundle#end()

filetype plugin indent on
```

Problem was at the first row - as soon as I changed it to `filetype off` without any other changes then Elixir highlighting started to work! Thanks again for all the help and making me understand slightly better how filetype detection work in vim. I've never had any need to understand that part of Vim until now.

I'm closing this bug and hopefully someone else having this exact problem can find solution from here.
